containers/ecr-viewer/tests/unit/app/components/Filters.test.tsx

```diff
@@ -21,471 +22,329 @@ const MOCK_PROPS = {
+        Promise.resolve({
+          ok: true,
+          json: () => Promise.resolve(conditions),
+        } as unknown as Response),
+      );
     });
-    await user.click(radio);
-    expect(radio).toBeChecked();
-
-    await user.click(toggleButton);
 
-    // Selection should not persist because filter was not applied
-    expect(screen.getByText("Last year")).toBeInTheDocument();
-    await user.click(toggleButton);
-    const radioAfterReset = screen.getByRole("radio", {
-      name: "Last 7 days",
+    it("renders correctly after opening conditions filter box", async () => {
+      const user = userEvent.setup();
+      const { container } = renderFilters(conditions);
+
+      const toggleFilterButton = screen.getByRole("button", {
+        name: /Filter by reportable condition/i,
+      });
+      await user.click(toggleFilterButton);
+
+      expect(container).toMatchSnapshot();
+    });
+
+    it("Toggles filter by conditions combo box visibility", async () => {
+      const user = userEvent.setup();
+      renderFilters(conditions);
+      const toggleButton = await screen.findByRole("button", {
+        name: /Filter by reportable condition/i,
+      });
+
+      // Initially closed
+      expect(screen.queryByText(/Filter by reportable condition/)).toBeNull();
+
+      // Open on click
+      await user.click(toggleButton);
+      expect(
+        screen.getByText(/Filter by reportable condition/),
+      ).toBeInTheDocument();
+      const applyFilterButton = screen.getByRole("button", {
+        name: /Apply Filter/i,
+      });
+      expect(applyFilterButton).toBeInTheDocument();
+
+      // Close on click
+      await user.click(toggleButton);
+      expect(screen.queryByText(/Filter by reportable condition/)).toBeNull();
+    });
+
+    it("Fetches conditions on Filters component mount", async () => {
+      const user = userEvent.setup();
+      renderFilters(conditions);
+      const toggleButton = screen.getByRole("button", {
+        name: /Filter by reportable condition/i,
+      });
+
+      await user.click(toggleButton);
+
+      // Should have correct number of checkboxes
+      const checkboxes = screen.getAllByRole("checkbox");
+      expect(checkboxes).toHaveLength(conditions.length + 1); // conditions + select all
+
+      // Conditions should be listed
+      for (const condition of conditions) {
+        expect(await screen.findByLabelText(condition)).toBeInTheDocument();
+      }
+    });
+
+    it("updates filterConditions state when a checkbox is checked and unchecked", async () => {
+      const user = userEvent.setup();
+      renderFilters(conditions);
+      const toggleFilterButton = screen.getByRole("button", {
+        name: /Filter by reportable condition/i,
+      });
+
+      await user.click(toggleFilterButton);
+
+      //--------- UNCHECKING BUTTON
+      // Checkbox should initialize as checked
+      if (expectedElements.regularConditions) {
```

Comment:
looking at the code, the tag should count the no conditions reported as well - remove this if?